keystrokes/src/components/Character/CharacterDetails/ManualKeybindInputs.js

```diff
@@ -0,0 +1,368 @@
+					onChange={event => {
+						setKeybinding({
+							Spell: event.target.value,
+							Target: null,
+							Mod: null,
+							Key: null,
+						});
+						setInvalidBind([]);
+					}}
+				>
+					{Object.entries(Spells)
+						.filter(spell => spell[1].spec.includes(spec))
+						.map(spell => {
+							const existingSpellBinds = allKeybinds.filter(
+								bind => bind.Spell === spell[0]
+							);
+							return (
+								<MenuItem
+									key={spell[0]}
+									id={`${spell[0].replace(/ /g, '')}-option`}
+									value={spell[0]}
+								>
+									{keybinding.Spell !== spell[0] ? (
+										<Grid
+											container
+											direction="row"
+											justify="space-between"
+										>
+											<Grid item>{spell[0]}</Grid>
+											<Grid item>
+												<Tooltip
+													placement="right-end"
+													id={`${spell[0].replace(
+														/ /g,
+														''
+													)}-popup`}
+													title={
+														existingSpellBinds &&
+														existingSpellBinds.length ? (
+															<React.Fragment>
+																<Typography>
+																	Current
+																	keybinds for{' '}
+																	{spell[0]}
+																</Typography>
+																<ul>
+																	{existingSpellBinds.map(
+																		bind => (
+																			<li
+																				key={
+																					bind.Target
+																				}
+																			>
+																				<Typography>{`${bind.Target} ${bind.Mod} ${bind.Key}`}</Typography>
+																			</li>
+																		)
+																	)}
+																</ul>
+															</React.Fragment>
+														) : (
+															<React.Fragment>
+																<Typography>{`No keybinds yet for ${spell[0]}!`}</Typography>
+															</React.Fragment>
+														)
+													}
+												>
+													<Avatar
+														className={
+															existingSpellBinds &&
+															existingSpellBinds.length
+																? classes.keybindsComplete
+																: classes.keybindsIncomplete
+														}
+														id={`${spell[0].replace(
+															/ /g,
+															''
+														)}-status`}
+													>
+														<Typography variant="subtitle2">
+															{existingSpellBinds &&
+																existingSpellBinds.length}
+														</Typography>
+													</Avatar>
+												</Tooltip>
+											</Grid>
+										</Grid>
+									) : (
+										spell[0]
+									)}
+								</MenuItem>
+							);
+						})}
+				</TextField>
+			</Grid>
+			<Grid item className={classes.keybindingOptions}>
+				<TextField
+					className={classes.button}
+					select={!!keybinding.Spell}
+					disabled={!keybinding.Spell}
+					id={'target-selector'}
+					variant="outlined"
+					value={keybinding.Target || ''}
+					label="Target"
+					onChange={event => {
+						setKeybinding({
+							...keybinding,
```

Comment:
Dive these objects are the same might be worth creating it once and passing to both sets

Reply:
good call